torchmetrics/detection/map.py

```diff
@@ -279,20 +291,339 @@ def update(self, preds: List[Dict[str, Tensor]], target: List[Dict[str, Tensor]]
+            "gtIgnore": gt_ignore,
+            "dtIgnore": dt_ignore,
+        }
+
+    def _summarize(
+        self,
+        results: Dict,
+        ap: bool = True,
+        iou_threshold: Optional[float] = None,
+        area_range: str = "all",
+        max_dets: int = 100,
+    ) -> Tensor:
+        """Perform evaluation for single class and image.
+
+        Args:
+            results:
+                Dictionary including precision, recall and scores for all combinations.
+            ap:
+                Calculate average precision. Else calculate average recall.
+                default: `True`
+            iou_threshold:
+                IoU threshold. If set to `None` it all values are used. Else results are filtered.
+            area_range:
+                Bounding box area range key.
+                default: ``all``
+            max_dets:
+                Maximum detections.
+                default: ``100``
+        """
+        aind = [i for i, aRng in enumerate(self.bbox_area_ranges.keys()) if aRng == area_range]
+        mind = [i for i, mDet in enumerate(self.max_detection_thresholds) if mDet == max_dets]
+        if ap:
+            # dimension of precision: [TxRxKxAxM]
+            s = results["precision"]
+            # IoU
+            if iou_threshold is not None:
+                t = torch.where(iou_threshold == self.iou_thresholds)[0]
+                s = s[t]
+            s = s[:, :, :, aind, mind]
+        else:
+            # dimension of recall: [TxKxAxM]
+            s = results["recall"]
+            if iou_threshold is not None:
+                t = torch.where(iou_threshold == self.iou_thresholds)[0]
+                s = s[t]
+            s = s[:, :, aind, mind]
+        if len(s[s > -1]) == 0:
+            mean_s = torch.Tensor([-1])
+        else:
+            mean_s = torch.mean(s[s > -1])
+
+        return mean_s
+
+    def _calculate(self, class_ids: List) -> Tuple[Dict, MAPMetricResults, MARMetricResults]:
+        """Calculate the precision, recall and scores for all supplied label classes to calculate mAP/mAR.
+
+        Args:
+            class_ids:
+                List of label class Ids.
+        """
+        img_ids = torch.arange(len(self.groundtruth_boxes), dtype=torch.int).tolist()
+
+        maxDetections = self.max_detection_thresholds[-1]
+        area_ranges = self.bbox_area_ranges.values()
+
+        ious = {
+            (id, class_id): self._compute_iou(id, class_id, maxDetections) for id in img_ids for class_id in class_ids
+        }
+
+        evalImgs = [
+            self._evaluate_image(id, class_id, area, maxDetections, ious)
+            for class_id in class_ids
+            for area in area_ranges
+            for id in img_ids
+        ]
+
+        T = len(self.iou_thresholds)
+        R = len(self.rec_thresholds)
+        K = len(class_ids)
+        A = len(self.bbox_area_ranges)
+        M = len(self.max_detection_thresholds)
+        I = len(img_ids)  # noqa: E741
```

Comment:
lets avoid using capital letters fro variables, capitals are reserved for class names :]